### PR TITLE
add integration test and s3 ui test to circleci

### PR DESCRIPTION
**Pull request by sunchunqiang**
run integration test on CircleCI for aws mobile client module 
run s3 ui test on CircleCI

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -41,22 +41,39 @@ commands:
...
+          command: 
+            adb logcat > logcat.log
+          background: true
+      - run:
+          name: Wait emulator
+          command: |
+            echo "wait for emulator to have booted"
+            circle-android wait-for-boot
+            echo "unlock the emulator screen"
+            sleep 30
```
Comment (minbi) [inline .circleci/config.yml]: Can this be a loop that checks if the emulator is ready? Say, every 5 seconds.

File: .circleci/config.yml [hunk 2]
```diff
@@ -325,18 +341,162 @@ jobs:
...
+                - develop
+      - uitest:
           requires:
             - build
+          filters:
+            branches:
+              only:
+                - master
+                - develop 
+                - integrationtest
```
Comment (minbi) [inline .circleci/config.yml]: Remove

File: CircleciScripts/functions.py [hunk 3]
```diff
@@ -75,5 +76,15 @@ def getmodules(root):
...
     return modules ;
 
+def replacefiles(root, replaces):
+    for replaceaction in replaces:
+        match = replaceaction["match"]
+        replace = replaceaction["replace"]
+        files = replaceaction["files"]
+        for file in files:
+            targetfile = os.path.join(root, file)
+            runcommand(command = "sed -r -i'' 's/{0}/{1}/' '{2}'".format(match, replace, targetfile), logcommandline = True)
```
Comment (minbi) [inline CircleciScripts/functions.py]: This needs documentation

File: aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientOfflineTest.java [hunk 4]
```diff
@@ -113,6 +115,7 @@ public void testGetConfiguration() throws JSONException {
         }
     }
 
+    @Ignore("This test may get stuck")
```
Comment (minbi) [inline aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientOfflineTest.java]: This needs to be documented that the network conditions might be switched during test run causing the test to fail.

File: aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientTest.java [hunk 5]
```diff
@@ -356,7 +358,7 @@ public void onUserStateChanged(UserStateDetails details) {
         countDownLatch.await(5, TimeUnit.SECONDS);
         assertFalse(triggered.get());
     }
-
+    @Ignore("This test case may get other test cases stuck")
```
Comment (minbi) [inline aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientTest.java]: Causes current available emulators to crash

File: aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientTestBase.java [hunk 6]
```diff
@@ -49,4 +50,9 @@ public static String getValidJWT(long expiryInSecs){
         String validAccessToken = accessToken_p1_Base64+"."+accessToken_p2_Base64+"."+accessToken_p3_Base64;
         return validAccessToken;
     }
+    @Test
+    public void empty()
```
Comment (minbi) [inline aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientTestBase.java]: Document that this is need for CircleCI to run the test suite.

File: .circleci/config.yml (hunk 1 above)
Comment (sunchunqiang) [reply .circleci/config.yml]: done !

File: CircleciScripts/functions.py (hunk 3 above)
Comment (sunchunqiang) [reply CircleciScripts/functions.py]: Done

File: aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientOfflineTest.java (hunk 4 above)
Comment (sunchunqiang) [reply aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientOfflineTest.java]: done

File: aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientTestBase.java (hunk 6 above)
Comment (sunchunqiang) [reply aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientTestBase.java]: done

File: CircleciScripts/unlock_emulatorscreen.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+from functions import runcommand
+import time
+int ncount = 0 
+print("unlocking emulator screen ...")
+while True:
```
Comment (minbi) [inline CircleciScripts/unlock_emulatorscreen.py]: Should this fail after some time or are we ok with this running until CircleCI times out?

Comment (sunchunqiang) [reply CircleciScripts/unlock_emulatorscreen.py]: I  believe we need let this fail after some time

**Comment by sunchunqiang** [comment]
merged to mainline

**Pull request closed without merging**